### PR TITLE
Parse the diagnostics to check for errors during the self_test

### PR DESCRIPTION
**Pull request by ugocupcic**
Parses the diagnostics aggregated topic. Check if some values are within a certain interval (for example jitter), or simply if the status is OK (for example for the Bridge).

To test it: make test

### CODE REVIEW

File: sr_self_test/include/sr_self_test/diagnostic_parser.hpp [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    ~MinMaxDiagnostics()
+    {};
+
+    virtual void parse_diagnostics(std::vector<diagnostic_msgs::KeyValue> values,
+                                   short level)
+    {
+      for( size_t values_i = 0; values_i < values.size(); ++values_i )
+      {
+        DiagMap::iterator values_it;
+        values_it = values_->begin();
```
Comment (markpitchless) [inline sr_self_test/include/sr_self_test/diagnostic_parser.hpp:121]: You don't need this line, as the same thing happens in the for loop below. Could then also pull the declaration into the for (although they get a little long this way).

```
for(DiagMap::iterator values_it = values_->begin(); values_it != values_->end(); ++values_it) 
```

File: sr_self_test/src/diagnostic_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    diag_sub_ = nh_.subscribe("diagnostics_agg", 1, &DiagnosticParser::diagnostics_agg_cb_, this);
+  }
+
+  void DiagnosticParser::parse_diagnostics(diagnostic_updater::DiagnosticStatusWrapper& status)
+  {
+    for(size_t i=0; i<50; ++i)
+    {
+      ros::Duration(0.1).sleep();
+      ros::spinOnce();
```
Comment (markpitchless) [inline sr_self_test/src/diagnostic_parser.cpp:46]: Why do you need to spin once here? This block could do with a comment.

File: sr_self_test/src/sr_self_test.cpp [hunk 3]
```diff
@@ -262,6 +262,15 @@
     safe_target.joint_target = 0.0;
     safe_targets_->insert( std::pair<std::string, sr_robot_msgs::joint>(safe_target.joint_name, safe_target) );
   }
+
+
+  ///////
+  // PARSING DIAGNOSTICS
+
+  void SrSelfTest::parse_diagnostics_()
```
Comment (markpitchless) [inline sr_self_test/src/sr_self_test.cpp:270]: Don't think this method is needed just seems to be wrapping add_diagnostic_parser with a different name. Caller could just do:

```
test_runner_.add_diagnostic_parser(); 
```

File: sr_self_test/src/diagnostic_parser.cpp (hunk 2 above)
Comment (ugocupcic) [inline sr_self_test/src/diagnostic_parser.cpp:46]: If you don't spin once here you wouldn't get the callback for the message.

**Comment by ugocupcic** [comment]
all fixed in B_self_tests_small_fixes

**Pull request merged**